receiver/oracledbreceiver/factory.go

```diff
@@ -32,13 +38,48 @@ func NewFactory() component.ReceiverFactory {
 	return component.NewReceiverFactory(
 		typeStr,
 		createDefaultConfig,
-		component.WithMetricsReceiver(createMetricsReceiver, stability))
+		component.WithMetricsReceiver(createReceiverFunc(func(dataSourceName string) (*sql.DB, error) {
+			return sql.Open("oracle", dataSourceName)
+		}, newDbClient), stability))
 }
 
-func createMetricsReceiver(ctx context.Context, settings component.ReceiverCreateSettings, receiver component.Config, metrics consumer.Metrics) (component.MetricsReceiver, error) {
-	return &oracledbreceiver{}, nil
+func createDefaultConfig() component.Config {
+	return &Config{
+		ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr)),
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			ReceiverSettings:   config.NewReceiverSettings(component.NewID(typeStr)),
+			CollectionInterval: 10 * time.Second,
+		},
+		MetricsSettings: metadata.DefaultMetricsSettings(),
+	}
 }
 
-func createDefaultConfig() component.Config {
-	return &Config{ReceiverSettings: config.NewReceiverSettings(component.NewID(typeStr))}
+type sqlOpenerFunc func(dataSourceName string) (*sql.DB, error)
+
+func createReceiverFunc(sqlOpenerFunc sqlOpenerFunc, clientProviderFunc clientProviderFunc) component.CreateMetricsReceiverFunc {
+	return func(
+		ctx context.Context,
+		settings component.ReceiverCreateSettings,
+		cfg component.Config,
+		consumer consumer.Metrics,
+	) (component.MetricsReceiver, error) {
+		sqlCfg := cfg.(*Config)
+		var opts []scraperhelper.ScraperControllerOption
+		metricsBuilder := metadata.NewMetricsBuilder(sqlCfg.MetricsSettings, settings.BuildInfo)
+		datasourceURL, _ := url.Parse(sqlCfg.DataSource)
+		instanceName := datasourceURL.Host
+
+		mp := newScraper(settings.ID, metricsBuilder, sqlCfg.MetricsSettings, sqlCfg.ScraperControllerSettings, settings.TelemetrySettings.Logger, func() (*sql.DB, error) {
+			return sqlOpenerFunc(sqlCfg.DataSource)
+		}, clientProviderFunc, instanceName)
+		opt := scraperhelper.AddScraper(mp)
+		opts = append(opts, opt)
+
+		return scraperhelper.NewScraperControllerReceiver(
+			&sqlCfg.ScraperControllerSettings,
+			settings,
+			consumer,
+			opts...,
```

Comment:
Looks like `opts` is not necessary
```suggestion
			scraperhelper.AddScraper(mp),
```